src/coreclr/vm/methodtable.cpp

```diff
@@ -7780,7 +7779,22 @@ PCODE MethodTable::GetRestoredSlot(DWORD slotNumber)
     //
 
     PCODE slot = GetCanonicalMethodTable()->GetSlot(slotNumber);
-    _ASSERTE(slot != NULL);
+#ifndef DACCESS_COMPILE
+    if (slot == (PCODE)NULL)
+    {
+        // This is a slot that has not been filled in yet. This can happen if we are
+        // looking at a slot which has not yet been given a temporary entry point.
+        MethodDesc *pMD = GetCanonicalMethodTable()->GetMethodDescForSlot_NoThrow(slotNumber);
```

Comment:
What is returned if it would throw?

Reply:
The same value. The difference is that the throwing variant works by filling in the slot and then computing the `MethodDesc` from that, and the NoThrow variant does a lookup for the MethodDesc but does not fill in the slot with a code pointer. I've added a comment to the NoThrow variant to this effect.